### PR TITLE
Add github install instructions

### LINKED ISSUE #24: Unable to install with npm
I tried to install shx with npm (globaly, locally, as dep, as devDep) both linux and windows machines. But it breaks during linking. It looks like it does not find `lib/cli.js` (see the log below).
```
npm install shelljs/shx
```

npm-debug.log
```
1095 verbose about to build /tmp/node_modules/shx/node_modules/shelljs
1096 info build /tmp/node_modules/shx/node_modules/shelljs
1097 info linkStuff shelljs@0.6.0
1098 silly linkStuff shelljs@0.6.0 has /tmp/node_modules/shx/node_modules as its parent node_modules
1099 verbose linkBins shelljs@0.6.0
1100 verbose link bins [ { shjs: './bin/shjs' },
1100 verbose link bins   '/tmp/node_modules/shx/node_modules/.bin',
1100 verbose link bins   false ]
1101 verbose linkMans shelljs@0.6.0
1102 verbose rebuildBundles shelljs@0.6.0
1103 verbose rebuildBundles [ 'glob', 'interpret', 'rechoir' ]
1104 silly gentlyRm /tmp/node_modules/shx/node_modules/.bin/shjs is being purged
1105 verbose gentlyRm don't care about contents; nuking /tmp/node_modules/shx/node_modules/.bin/shjs
1106 info install shelljs@0.6.0
1107 info postinstall shelljs@0.6.0
1108 verbose unlock done using /home/kirly/.npm/_locks/shelljs-4afc8c83066b6587.lock for /tmp/node_modules/shx/node_modules/shelljs
1109 verbose about to build /tmp/node_modules/shx
1110 info build /tmp/node_modules/shx
1111 info linkStuff shx@0.0.1
1112 silly linkStuff shx@0.0.1 has /tmp/node_modules as its parent node_modules
1113 verbose linkBins shx@0.0.1
1114 verbose link bins [ { shx: 'lib/cli.js' }, '/tmp/node_modules/.bin', false ]
1115 verbose linkMans shx@0.0.1
1116 verbose rebuildBundles shx@0.0.1
1117 verbose rebuildBundles [ '.bin', 'shelljs' ]
1118 verbose unlock done using /home/kirly/.npm/_locks/shx-f2158f07ac2468a2.lock for /tmp/node_modules/shx
1119 verbose stack Error: ENOENT, chmod '/tmp/node_modules/shx/lib/cli.js'
1119 verbose stack     at Error (native)
1120 verbose cwd /tmp
1121 error Linux 3.16.0-4-amd64
1122 error argv "/usr/bin/node" "/usr/bin/npm" "i" "-D" "shelljs/shx"
1123 error node v0.12.8
1124 error npm  v2.14.9
1125 error path /tmp/node_modules/shx/lib/cli.js
1126 error code ENOENT
1127 error errno -2
1128 error enoent ENOENT, chmod '/tmp/node_modules/shx/lib/cli.js'
1128 error enoent This is most likely not a problem with npm itself
1128 error enoent and is related to npm not being able to find a file.
1129 verbose exit [ -2, true ]
1130 verbose unbuild node_modules/shx
1131 info preuninstall shx@0.0.1
1132 info uninstall shx@0.0.1
1133 verbose unbuild rmStuff shx@0.0.1 from /tmp/node_modules
1134 silly gentlyRm /tmp/node_modules/.bin/shx is being gently removed
1135 silly gentlyRm verifying /tmp is an npm working directory
1136 silly gentlyRm containing path /tmp is under npm's control, in /tmp
1137 silly gentlyRm deletion target /tmp/node_modules/.bin/shx is under /tmp
1138 verbose gentlyRm vacuuming from /tmp/node_modules/.bin/shx up to /tmp
1139 info postuninstall shx@0.0.1
1140 silly gentlyRm /tmp/node_modules/shx is being purged from base /tmp
1141 verbose gentlyRm don't care about contents; nuking /tmp/node_modules/shx
1142 silly vacuum-fs purging /tmp/node_modules/shx
1143 silly vacuum-fs removing /tmp/node_modules
1144 silly vacuum-fs finished vacuuming up to /tmp
```

### PR DESCRIPTION
**Pull request by levithomason**
Fixes #24

### CODE REVIEW

**Comment by nfischer** [comment]
These commands fail though (correct me if I'm wrong), which (if I were following this guide) would surprise me. That either needs to be fixed or documented.

**Comment by levithomason** [comment]
See #24, confirmed everything works as it should.

File: README.md [hunk 1]
```diff
@@ -47,3 +47,16 @@ rm('foo.txt');
...
 
 And then calling it with `node scripts/scriptname.js`
+
+Installing from GitHub?
+----------------------
+
+If you are installing directly from GitHub, `npm install shelljs/shx`, you'll need to build `shx` locally.  This project does not commit built assets to the GitHub repo.  To build locally:
+
+```
+cd node_modules/shx
```
Comment (nfischer) [inline README.md]: How does the `node_modules/shx` directory get created? We should include that command above this. Here's what I tried (that didn't work):

```bash
$ npm install shelljs/shx # I'm expecting this to have errors
... Lots of errors ...
$ cd node_modules/shx/ # I wasn't expecting this command to fail
no such file or directory: node_modules/shx/
```

**Comment by levithomason** [comment]
Indeed, same error here.  It looks like `chmod` fails because there is no `lib` folder yet, and it is trying to change perms there.

I think the better solve here is to put the built assets on a `dist` branch during CI.  This way, there are no special steps needed to npm install from github, just install the built version from the `dist` branch.  I just pushed the built version there for testing.  Now I get:

```shell
› npm i shelljs/shx#dist

> shx@0.0.1 postinstall /Users/levithomason/src/stardust/node_modules/shx
> npm run build

> shx@0.0.1 prebuild /Users/levithomason/src/stardust/node_modules/shx
> rimraf lib

> shx@0.0.1 build /Users/levithomason/src/stardust/node_modules/shx
> babel src -d lib

src/cli.js -> lib/cli.js
src/printCmdRet.js -> lib/printCmdRet.js
src/shx.js -> lib/shx.js
stardust@0.13.3 /Users/levithomason/src/stardust
└── shx@0.0.1  extraneous (git://github.com/shelljs/shx.git#3b18c3f7132ecd5c802a2506b75057dadf0f22e5)
```

To get the built assets there on CI, it is pretty trivial:

```shell
git checkout dist
npm run build
git add -f lib
git commit -m 'chore(build): force add built assets'
git push
```

**Comment by ariporad** [comment]
@levithomason: I think we shouldn't maintain a dist branch, I don't see why it would be worth the effort. May as well publish to npm (even with a `next` tag).

If you'll forgive me for bringing this up again, this is one of the reasons I really like `semantic-release`, as soon as code is ready to go, it gets released. It also prevents build artifacts from ever being in git.

**Comment by nfischer** [comment]
I don't see a reason to have "install from github" instructions when we can't install from github. This should be closed I think.

**Comment by ariporad** [comment]
Agreed.

**Comment by levithomason** [comment]
Note, I've already created a `dist` branch and installed shx with npm from GitHub.  See above ^.  In either case, I agree, let's just have ppl install from NPM.

Closing and cleaning up.

**Pull request closed without merging**